app/renderer/reducers/ServerMonitor.js

```diff
@@ -37,13 +40,21 @@ export default function serverMonitor (state = initialState, action) {
         sessionId: action.sessionUUID,
       };
     case LOGS_RECEIVED:
+      // TODO: We should dump logs to a txt file that can be exported
+      var logLines = state.logLines.concat(action.logs.map((l) => {
```

Comment:
Yeah, case statements aren't blocks, so the variables are hoisted to the top of the switch statement, but uninitialized until the case it executed. This isn't so good.

The best solution would be to wrap the case stuff into a block.
```js
case LOGS_RECEIVED: {
  // TODO: We should dump logs to a txt file that can be exported
  let logLines = state.logLines.concat(action.logs.map((l) => {
    // attach a timestamp to each log line here when it comes in
    l.timestamp = moment().format('YYYY-MM-DD hh:mm:ss');
    return l;
  }));

  // Don't log more than MAX_LOG_LINES
  if (logLines.length > MAX_LOG_LINES) {
    logLines = logLines.slice(logLines.length - MAX_LOG_LINES);
  }

  return {
    ...state,
    logLines,
    serverStatus: STATUS_RUNNING
  };
}
```